### ISSUE TITLE
The problems about t-SNE visualization.

### ISSUE THREAD

**Issue by zhouyuan888888** [ShuoYang-1998/ICLR2021-Oral_Distribution_Calibration#9]
Hi, when I try to reproduce the t-SNE visualization as in the figure 2 of the original paper, I find I cannot reproduce the results, which is shown below:
![t_sne_1 (1)](https://user-images.githubusercontent.com/41813767/105626406-52956980-5e6a-11eb-904c-55580605f204.png)

The corresponding code is by adding t-SNE visualization after the original code: 
       ##original codes###
       sampled_data = np.concatenate([sampled_data[:]]).reshape(n_ways * n_shot * num_sampled, -1)
        X_aug = np.concatenate([support_data, sampled_data])
        Y_aug = np.concatenate([support_label, sampled_label])
        
        ##add t-SNE visualization to analyze##
        tsne = TSNE(n_components=2, init="pca", random_state=0)
        feats_tsne = tsne.fit_transform(X_aug)
        fig = plot_embedding(feats_tsne, Y_aug, "t-SNE")
        if not os.path.exists("./figure/"):
            os.mkdir("./figure/")
        plt.savefig("./figure/t_sne_{}".format(n_runs))

What's wrong with my codes T_T?

**Comment by zhouyuan888888** [comment]
The details of the function plot_embedding:
`def plot_embedding(data, label, title):
    x_min, x_max = np.min(data, 0), np.max(data, 0)
    data = (data - x_min) / (x_max - x_min)
    np.random.seed(0)

    length = data.shape[0]
    R = np.random.rand(np.max(label))
    G = np.random.rand(np.max(label))
    B = np.random.rand(np.max(label))

    fig = plt.figure()
    ax = plt.subplot(111)
    for i in range(length):
        if  i < 5:
            plt.text(data[i, 0], data[i, 1], str(label[i]),color="red", fontdict={'weight': 'bold', 'size': 25})
        else:
            plt.text(data[i, 0], data[i, 1], str(label[i]), color=(R[label[i] - 1], G[label[i] - 1], B[label[i] - 1]), fontdict={'weight': 'bold', 'size': 9})

    #plt.xticks([])
    #plt.yticks([])
    plt.title(title)
    return fig`

**Comment by ShuoYang-1998** [comment]
Hi, when you try to visualize the generated features, you can set the \alpha added on the covariance matrix as zero.

**Comment by zhouyuan888888** [comment]
Thank you for your quick reply^_^! Do you choose cosine similarity to measure the sample pairs' distance in the t-SNE visualization?
